lib/Doctrine/ORM/Mapping/NamingStrategy.php

```diff
@@ -15,64 +15,46 @@ interface NamingStrategy
      * Returns a table name for an entity class.
      *
      * @param class-string $className
-     *
-     * @return string A table name.
      */
-    public function classToTableName($className);
+    public function classToTableName(string $className): string;
 
     /**
      * Returns a column name for a property.
      *
-     * @param string       $propertyName A property name.
-     * @param class-string $className    The fully-qualified class name.
-     *
-     * @return string A column name.
+     * @param class-string $className
      */
-    public function propertyToColumnName($propertyName, $className = null);
+    public function propertyToColumnName(string $propertyName, string $className): string;
 
     /**
      * Returns a column name for an embedded property.
      *
-     * @param string       $propertyName
-     * @param string       $embeddedColumnName
      * @param class-string $className
      * @param class-string $embeddedClassName
-     *
-     * @return string
      */
     public function embeddedFieldToColumnName(
-        $propertyName,
-        $embeddedColumnName,
-        $className = null,
-        $embeddedClassName = null
-    );
+        string $propertyName,
+        string $embeddedColumnName,
+        string $className,
+        string $embeddedClassName,
+    ): string;
 
     /**
      * Returns the default reference column name.
-     *
-     * @return string A column name.
      */
-    public function referenceColumnName();
+    public function referenceColumnName(): string;
 
     /**
      * Returns a join column name for a property.
-     *
-     * @param string $propertyName A property name.
-     *
-     * @return string A join column name.
      */
-    public function joinColumnName($propertyName/*, $className = null */);
+    public function joinColumnName(string $propertyName/*, $className = null */): string;
```

Comment:
Isn't $className supposed to  be added in 3.0?

Reply:
Yes, but I was gonna do that in a separate PR (#9755)